themes/default/content/blog/organizing-aws-accounts-with-pulumi/index.md

```diff
@@ -0,0 +1,290 @@
+
+**Note**: If you are using one of the supported CI/CD services you might be able to use federated credentials using OIDC. So if you are going to use that auth mechanism, you can skip creation of the fixed automation user account since you will not need its
+credentials.
+
+## Organizational Unit (OU)
+
+An org unit can represent anything. In our example it represents the development team. You could also have multiple OUs and even nest them to create sub-hierarchies, depending on your use-case.
+
+The `organizations` module in the AWS provider package contains the resources we need in order to create an OU and an account.
+
+```typescript
+const devOrgUnit = new aws.organizations.OrganizationalUnit("orgUnit", {
+   parentId: organization.then((o) => o.roots[0].id),
+   name: "Development",
+});
+
+const devAccount = new aws.organizations.Account(
+   "devAccount",
+   {
+       name: "DeveloperAccount",
+       parentId: devOrgUnit.id,
+       email: devAccountEmailContact,
+	// This role can be used by this Pulumi app to execute
+	// an AssumeRole action on this new member account.
+       roleName: "OrganizationalAccountAccessRole",
+       // IMPORTANT! Set this to `false` if you do not wish to have
+       // accounts closed when the account resource is removed from
+       // your Pulumi app.
+       closeOnDeletion: true,
+   },
+   { protect: true }
+);
+```
+
+{{% notes type="info" %}}
+An important note about the `Account` resource and destroying them. Notice how we set the `closeOnDeletion` to `true`. Closing accounts on deletion will only succeed if you have not exceeded the quotas set by AWS for account closure. So set that flag accordingly. If you expect to be closing many accounts frequently (perhaps for an ephemeral use-case), you may want to set the flag to `false` so you can close them manually or have some other sort of asynchronous automation do it for you. Otherwise, your Pulumi execution will fail when AWS returns an error code due to the quota being exceeded.
+{{% /notes %}}
+
+## Developers’ Access
+
+We’ll create an IAM group for developers and assign a group policy that allows its members to perform the `sts:AssumeRole` on any AWS account but only using a specific role.
+
+```typescript
+const devGroup = new aws.iam.Group("developers", {
+   name: "developers",
+});
+
+new aws.iam.GroupPolicy("developersGroupPolicy", {
+   group: devGroup.name,
+   policy: {
+       Statement: [
+           {
+               Effect: "Allow",
+               Action: "sts:AssumeRole",
+               Resource: `arn:aws:iam::*:role/DeveloperRole`,
```

Comment:
Also here -- might be good to just mention briefly that these (this one and `AutomationRole`) are AWS built-ins.

Reply:
This role and the `AutomationRole` are roles that the app creates. In the actual example I have these in two variables. I replaced the variable references with hard-coded strings for the sake of keeping things simple in the code snippet in this post. The relevant lines in the actual example: https://github.com/pulumi/examples/pull/1330/files#diff-b8897f56ce67084e1d7fd95fb1af74032ef404fee05abbb4ab013340f384b6a6R18-R19

Reply:
But I'll add a comment that these are roles created by the component resource for account permissions.